modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc

```diff
@@ -241,7 +241,7 @@ function _govcms_ckan_media_field_widget_visualisation_options() {
  * @return array
  *   Form structure for new elements.
  */
-function govcms_ckan_media_visualisation_default_key_config($form, $form_state, $config = array()) {
+function govcms_ckan_media_visualisation_default_key_config($form, $form_state, $config = array(), $key_type = 'numeric') {
```

Comment:
My 2c, I Agree with James, it is cleaner and configurations and settings are kind of one in the same.